api/app/serializers/spree/v2/storefront/shipment_serializer.rb

```diff
@@ -4,8 +4,12 @@ module Storefront
       class ShipmentSerializer < BaseSerializer
         set_type :shipment
 
-        attributes :tracking, :number, :cost, :display_cost, :discounted_cost, :display_discounted_cost,
-                   :final_price, :display_final_price, :item_cost, :display_item_cost, :shipped_at
+        attributes :number, :final_price, :display_final_price,
+                   :state, :shipped_at, :tracking_url
+
+        attribute :free, &:free?
+
+        has_many :shipping_rates, if: proc { |_record, params| params && params[:show_rates] == true }
```

Comment:
Maybe we could write `params&.dig(:show_rates)` or `params&.fetch(:show_rates, false)` here?

Reply:
fixed!